### ISSUE TITLE
TypeScript error on pullstate 1.11.3

### ISSUE THREAD

**Issue by andraaspar** [lostpebble/pullstate#38]
I'm getting the following TypeScript error when trying to compile a project:

```
ERROR in C:/.../node_modules/pullstate/dist/PullstateCore.d.ts(27,44):
TS2344: Type 'PullstateSingleton<S>' does not satisfy the constraint 'PullstateSingleton<IPullstateAllStores>'.
ERROR in C:/.../node_modules/pullstate/dist/PullstateCore.d.ts(27,92):
TS2344: Type 'PullstateSingleton<S>' does not satisfy the constraint 'PullstateSingleton<IPullstateAllStores>'.
ERROR in C:/.../node_modules/pullstate/dist/PullstateCore.d.ts(28,41):
TS2344: Type 'PullstateSingleton<S>' does not satisfy the constraint 'PullstateSingleton<IPullstateAllStores>'.
```

**Comment by andraaspar** [comment]
My tsconfig.json is pretty straightforward:

```json
{
  "compilerOptions": {
    "target": "ES2019",
    "module": "es2015",
    "lib": [
      "DOM",
      "DOM.Iterable",
      "ES2019"
    ],
    "jsx": "react",
    "resolveJsonModule": true,
    "strict": true,
    "moduleResolution": "node",
    "forceConsistentCasingInFileNames": true
  }
}
```

**Comment by lostpebble** [comment]
Someone else ran into strict type issues, they sent through a pull request which should have fixed it. https://github.com/lostpebble/pullstate/pull/37

I'll release a new version now with the changes.

**Comment by lostpebble** [comment]
Released the latest type changes in `1.12.0` please check it out and let me know if that fixes it for you. Unfortunately I don't have any projects currently using `strict: true` to make sure that it works completely - but since `pullstate` itself has been made strict now, I assume it should be fine.

**Comment by andraaspar** [comment]
Thanks for the quick response! Yes, that fixed it. Closing issue.
